README_fr-FR.md

```diff
@@ -0,0 +1,443 @@
+"pourquoi" vous faites ce changement contrairement à "comment"
+(le code explique déjà cela).
+Est-ce qu'il y a des effets ou d'autres conséquences qui ne sont
+pas intuitives concernant ce changement ? C'est l'endroit où vous
+pouvez les décrire.
+
+Les autres paragraphes viennent après des lignes vides.
+
+ - Les puces sont ok.
+
+ - Vous pouvez utilisez des astérisques et des puces, tout en gardant
+   une syntaxe correcte.
+
+Si vous utilisez un outil de suivi des problèmes,
+mettez des références à la fin comme ceci:
+
+Resolves: #123
+See also: #456, #789
+```
+
+## Rebase vs. Merge
+
+Cette section est un **TL;DR** de l'excellent tutoriel d'Atlassian, ["Merging vs. Rebasing"](https://www.atlassian.com/git/tutorials/merging-vs-rebasing).
+
+![](https://wac-cdn.atlassian.com/dam/jcr:01b0b04e-64f3-4659-af21-c4d86bc7cb0b/01.svg?cdnVersion=hq)
+
+### Rebase
+
+**TL;DR:** Applique vos commits de branche, un par un, sur la branche principale, générant un nouvel "arbre"
+
+![](https://wac-cdn.atlassian.com/dam/jcr:5b153a22-38be-40d0-aec8-5f2fffc771e5/03.svg?cdnVersion=hq)
+
+### Merge
+
+**TL;DR:** Créé un nouveau commit, appelé un _merge commit_, avec les différences entre ces deux branches.
+
+![](https://wac-cdn.atlassian.com/dam/jcr:e229fef6-2c2f-4a4f-b270-e1e1baa94055/02.svg?cdnVersion=hq)
+
+### Pourquoi certaines personnes préfères rebase plutôt que merge ?
+
+Je préfère particulièrement rebase plutôt que merge. Les raisons sont:
+
+* Cela génère un historique "propre", sans merge commits inutiles.
+* _Ce que vous voyez est ce que avez_, i.e., dans une code review tout les changements viennent d'un commit spécifique, empéchant les changements cachés dans les merge commits
+* Plus de merges sont résolus par la personne qui commit, et chaque merge change est dans un commit avec un message propre à celui-ci.
+    * Ce n'est pas commun de chercher et d'examiner les merge commits, donc les éviter assure que tous les changements ont un commit où ils sont.
+
+### Quand "squash"
+
+"Squasher" c'est prendre une série de commits et les condensser dans un seul commit.
+
+C'est utile dans plusieurs situations, e.g.:
+
+- Réduire le nombre de commit avec peu ou pas de contexte (correction de fautes d'orthographe, formatage, choses oubliées)
+- Regrouper des changements séparés, qui ont plus de sens une fois regroupés.
+- Réécrire les commits de _travail en cours_
+
+### Quand éviter rebase ou squash ?
+
+Il faut éviter rebase et squash dans les commits publiques ou dans les branches partagées où plusieurs personnes travaillent dessus.
+
+Rebase et squash réécrivent l'history and écrasent less commits existants, le faire sur des commits qui sont dans des branches partagées (i.e., les commits push sur le remote repository ou qui viennent d'autres branches) peut entrainer de la confusion, et les gens risquent de perdre leurs changements (autant localement que remotely) car les arbres et les conflicts divergent.
```

Comment:
`history` > `historique`
`and` > `et`
`less` > `les`
`remote repository` > `dépôt distant`
`remotely` > `à distance`